It generates a warning with GNU libc:

	% make clean all
	rm -f grep
	cc -std=c99 -pedantic -Wall -O2  -D_GNU_SOURCE=1 -o grep *.c
	fts.c:54:9: warning: 'MAX' macro redefined [-Wmacro-redefined]
	#define MAX(a,b) ((a)>(b)?(a):(b))
			^
	/usr/include/sys/param.h:103:9: note: previous definition is here
	#define MAX(a,b) (((a)>(b))?(a):(b))
			^
	1 warning generated.

Changing that `!defined(HAVE_DECL_MAX) || (HAVE_DECL_MAX==0)` to `defined(HAVE_MAX)` should probably fix it?

Also got some trailing whitespace in the Makefile.

--- a/Makefile
+++ b/Makefile
@@ -17,7 +17,6 @@ install:
 	
 	install -Dm644 grep.1 ${DESTDIR}${PREFIX}/share/man/man1/grep.1
 	install -Dm644 zgrep.1 ${DESTDIR}${PREFIX}/share/man/man1/zgrep.1
-	
 	cd ${DESTDIR}${PREFIX}/bin && ( \
 		ln -sf grep  egrep     ;\
 		ln -sf grep  fgrep     ;\
@@ -37,7 +36,6 @@ install:
 		ln -sf zgrep zstdegrep ;\
 		ln -sf zgrep zstdfgrep ;\
 	)
-	
 	cd ${DESTDIR}${PREFIX}/share/man/man1 && ( \
 		ln -sf grep.1 egrep.1      ;\
 		ln -sf grep.1 fgrep.1      ;\

--- a/fts.c
+++ b/fts.c
@@ -50,9 +50,7 @@ __RCSID("$NetBSD: fts.c,v 1.48 2015/01/29 15:55:21 manu Exp $");
 #include <string.h>
 #include <unistd.h>
 
-#if !defined(HAVE_DECL_MAX) || (HAVE_DECL_MAX==0)
-#define MAX(a,b) ((a)>(b)?(a):(b))
-#endif
+#define MAX_(a,b) ((a)>(b)?(a):(b))
 
 #if !defined(UINT_MAX) && (HAVE_DECL_UINTMAX_MAX==1)
 #define UINT_MAX UINTMAX_MAX
@@ -155,7 +153,7 @@ fts_open(char * const *argv, int options,
 	 * Start out with 1K of path space, and enough, in any case,
 	 * to hold the user's paths.
 	 */
-	if (fts_palloc(sp, MAX(fts_maxarglen(argv), MAXPATHLEN)))
+	if (fts_palloc(sp, MAX_(fts_maxarglen(argv), MAXPATHLEN)))
 		goto mem1;
 
 	/* Allocate/initialize root's parent. */

